tools/payments/report.go

```diff
@@ -53,6 +55,27 @@ func (r AttestedReport) EnsureUniqueDest() error {
 	return nil
 }
 
+// EnsureTransactionAmountsMatch checks each transaction by To address and errors if the Amounts do
+// not match between the AttestedReport and the provided PreparedReport.
+func (ar AttestedReport) EnsureTransactionAmountsMatch(pr PreparedReport) error {
+	preparedMap := make(map[string]decimal.Decimal, len(pr))
+	for _, paymentDetails := range pr {
+		preparedMap[paymentDetails.To] = paymentDetails.Amount
+	}
+	for _, attestedDetails := range ar {
+		if preparedMap[attestedDetails.To].Cmp(attestedDetails.Amount) != 0 {
```

Comment:
Suggestion: `Cmp` is helpful when you're interested in whether it's less, equal or greater in one step. Here, you're interested in equality, therefore a more readable option is to go with `Equal` which communicates the intent as clear is possible:

```suggestion
		if !preparedMap[attestedDetails.To].Equal(attestedDetails.Amount) {
```